README.md

```diff
@@ -929,15 +904,25 @@ Internal `viewType` property.
 
 ## Developing
 
-If you're cloning this repository in order to hack on it, you need to setup the `deltachat-core` submodule, before doing `npm install`.
-
-The following commands should be enough to get started.
+You can build against an uninstalled `deltachat-core` by using a few
+environment variables.  You need to point to both the `deltachat-core`
+repository and to its build directory using the `DC_SRC` and
+`DC_BUILD` environment variables:
 
 ```
-git clone https://github.com/deltachat/deltachat-node.git
-cd deltachat-node
-npm run submodule
+export DC_SRC=../deltachat-core
+export DC_BUILD=../deltachat-core/builddir
 npm install
+# or:
+#   node-gyp configure --verbose
+#   node-gyp build --verbose
+```
+
+Now you also need to set `LD_LIBRARY_PATH` to the built version before
```

Comment:
Not really a fan of the LD_LIBRARY_PATH thing... Is there no other solution to this?

Reply:
Yeah, this whole supporting uninstalled dc-core is really messy and i don't like any of it really.  The current procedure is much harder then doing `pushd ../dc-core; git checkout branch; meson build-my-branch; pushd build-my-branch; ninja; sudo ninja install; popd; popd; npm install`.

So can we ditch all this complexity?  Alternatively someone should do more gyp-build spelonking and figure out how to more easily do this.

Reply:
s/current procedure/suggested procedure in this PR/

Reply:
We can bake in an -Wl,-rpath in the ldflags of gyp to avoid LD_LIBRARY_PATH.

I think it's also possible to invoke `node-gyp configure` with custom vars which is nicer than env vars, but don't know yet

Reply:
@Jikstra check the current readme (and try it out!) for a much simpler version.  bindings.gyp is now also much more readable.  I may even somewhat like this approach, but it needs a small PR to dc-core to make it work (to put the `deltachat.h` file in the right location inside the builddir, you can fudge it by manually doing this to try it out).

Let me know if this is reasonable to you.

Reply:
I like it, didn't test it out but looks good. Is there a way to use relative paths? I think with `$ORIGIN` on linux this could be possible? Not being able to move the deltachat folder around is maybe okay but still a drawback.

Reply:
Could you elaborate what you're aiming for here?  Because the uninstalled libdeltachat.so lives in a different directory then you'd have to move two directories together.  Since this mechanism is to run against an in-dev version of libdeltachat.so I'm not sure what the usecase is to be able to move it around.

The last sentence is a bit more of a problem, I do not think deltachat-core should be bundled at all and that's explicitly what this PR is trying to do.  This repo with the node bindings is by definition being compiled against a certain node version.  Since this is generic extension-module source code the version of node itself or how it is built and linked is not under the control of this repo.  That makes it almost impossible [0] for this repo to know how to compile deltachat-core correctly in a way which will work.

[0] i.e. theoretically possible with lots of work and lots of config switches to cater for all possible variations we have thought off but practically this is a ton of difficult work and there will always be edge cases uncatered for